packages/contracts/test/plugin/plugin-setup-processor.ts

```diff
@@ -720,186 +882,164 @@ describe('Plugin Setup Processor', function () {
-        const pluginSetupV1 = pluginSetupV1Mock.address;
-
-        const {plugin, helpers, prepareInstallpermissions} =
-          await prepareInstallation(
-            psp,
-            daoAddress,
-            pluginSetupV1,
-            pluginSetupMockRepoAddress,
-            EMPTY_DATA
-          );
-
-        await psp.applyInstallation(
-          targetDao.address,
-          pluginSetupV1,
-          pluginSetupMockRepoAddress,
-          plugin,
-          prepareInstallpermissions
-        );
-
+      it('returns permissions and initData correctly', async () => {
         const pluginUpdateParams = {
-          plugin: plugin,
-          pluginSetupRepo: pluginSetupMockRepoAddress,
-          currentPluginSetup: pluginSetupV1,
-          newPluginSetup: pluginSetupV2Mock.address,
+          plugin: proxy,
+          pluginSetupRepo: repoUUPS.address,
+          currentPluginSetup: setupV1.address,
+          newPluginSetup: setupV2.address,
         };
 
         const result = await psp.callStatic.prepareUpdate(
-          daoAddress,
+          targetDao.address,
           pluginUpdateParams,
-          helpers,
+          helpersV1,
           EMPTY_DATA
         );
 
         const permissions = result[0];
         const initData = result[1];
 
-        expect(permissions.length).to.be.equal(2);
-        expect(initData).not.to.be.equal('');
+        expect(permissions).to.deep.equal(
+          mockPermissionsOperations(1, 2, Operation.Grant).map(perm =>
+            Object.values(perm)
+          )
+        );
+        expect(initData).not.to.be.equal(''); // TODO, improve test
       });
 
-      it('correctly prepares an update', async () => {
-        const daoAddress = targetDao.address;
-        const pluginSetupV1 = pluginSetupV1Mock.address;
-
-        const {plugin, helpers, prepareInstallpermissions} =
-          await prepareInstallation(
+      it('prepares an update', async () => {
+        await expect(
+          prepareUpdate(
             psp,
-            daoAddress,
-            pluginSetupV1,
-            pluginSetupMockRepoAddress,
+            targetDao.address,
+            proxy,
+            setupV1.address,
+            setupV2.address,
+            repoUUPS.address,
+            helpersV1,
             EMPTY_DATA
-          );
-
-        await psp.applyInstallation(
-          targetDao.address,
-          pluginSetupV1,
-          pluginSetupMockRepoAddress,
-          plugin,
-          prepareInstallpermissions
-        );
-
-        const pluginUpdateParams = {
-          plugin: plugin,
-          pluginSetupRepo: pluginSetupMockRepoAddress,
-          currentPluginSetup: pluginSetupV1,
-          newPluginSetup: pluginSetupV2Mock.address,
-        };
-
-        await expect(
-          psp.prepareUpdate(daoAddress, pluginUpdateParams, helpers, EMPTY_DATA)
-        ).to.emit(psp, EVENTS.UpdatePrepared);
+          )
+        ).to.not.be.reverted;
       });
```

Comment:
lacks test which reverts correctly if applyUpdate doesn't have ROOT permission for dao.

Reply:
Good catch. Added and I cleaned apply update tests a bit more.

Reply:
Good catch. Added it and cleaned applyUpdate tests a bit.